### PR TITLE
New Components Implemented

### PR DESCRIPTION
**Pull request by everton3x**
Implementation to:

TCalendar -> Calendar
TDateEdit -> InputDate
TTimeEdit -> InputTime
TSpinEdit/TFloatSpinEdit -> InputNumber
TEdit -> InputPassword

### CODE REVIEW

**Comment by reisraff** [comment]
Hey @everton3x I have one advice for you.

keep your branch master updated with the gabrielrcouto/master, and do your changes in separated branchs, that way you will avoid merge conflicts.

**Comment by everton3x** [comment]
Tanks @reisraff.

Actually, I'm a beginner in the contributions of GitHub. I only used it to store some of my projects.

If you have any material beyond the basics about Git I appreciate the suggestion. If it is in Portuguese, thank you, because my English is not very good (God save the Google Translate) What I read until today has always been rather superficial.

**Comment by alrik11es** [comment]
What happened to this PR (Seems really interesting) may I help with something?

**Comment by reisraff** [comment]
This PR has conflicts with the HEAD of master, that means @everton3x should rebase your code with the last update of gabrielrcouto/master, that way we can accept that, and merge ;)

**Comment by reisraff** [comment]
@everton3x can you please update this PR?

**Comment by everton3x** [comment]
From what I've seen, the problem is the size of some lines.

This week I'm pretty busy, but next week I can do it without problems.

**Comment by everton3x** [comment]
After a hard work correcting the source code to fit the PSR2, everything is ready. I'm still struggling to fit into PSR2.

**Comment by reisraff** [comment]
@everton3x okay, nice work, but still have some notes to review.

https://github.com/gabrielrcouto/php-gui/pull/113#discussion_r109942872
https://github.com/gabrielrcouto/php-gui/pull/113#discussion_r109944049
https://github.com/gabrielrcouto/php-gui/pull/113#discussion_r109944027
https://github.com/gabrielrcouto/php-gui/pull/113#discussion_r109943277
https://github.com/gabrielrcouto/php-gui/pull/113#discussion_r109943078

File: src/Components/InputDate.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        return $this->get('Text');
+    }
+
+    /**
+     * Returns a string with the representation [option1, option2, ...] to be configured by Lazarus.
+     *
+     * @param array $options the options
+     * @return string a string [option1, options2, option3,...]
+     */
+    private function getOptionString($options)
```
Comment (everton3x) [inline src/Components/InputDate.php]: Would it be convenient to move it to a trait, or an intermediate class that extends VisualObject to be then extended by InputDate and InputFile?

File: src/Components/InputDate.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        return "[$str]";
+    }
+
+    /**
+     * Parse an option string received from Lazarus.
+     *
+     * @param string $options the options string
+     * @return array
+     */
+    private function parseOptionString($options)
```
Comment (everton3x) [inline src/Components/InputDate.php]: Would it be convenient to move it to a trait, or an intermediate class that extends VisualObject to be then extended by InputDate and InputFile?

File: src/Components/InputDate.php (hunk 1 above)
Comment (reisraff) [inline src/Components/InputDate.php]: I guess a trait is better

File: src/Components/InputDate.php (hunk 2 above)
Comment (reisraff) [inline src/Components/InputDate.php]: Trait ;)

File: src/Components/InputDate.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    {
+        if (strlen($options) > 0) {
+            return array_map('trim', explode(',', $options));
+        } else {
```
Comment (reisraff) [inline src/Components/InputDate.php]: Apply here also object calisthenics.

**Comment by reisraff** [comment]
@everton3x do you get to finish this? is missing just some stuff... Or can I fork it and finish?

**Comment by everton3x** [comment]
@reisraff I started working on it a few days ago, but I had to stop work because of the professional priorities that have come up. If you have the will and availability, feel free to make a fork and finish. Thank you very much.

**Comment by reisraff** [comment]
Closing, continue with #117

**Pull request closed without merging**